packages/lesmis/src/views/DashboardReportTabView.js

```diff
@@ -39,8 +62,38 @@ const setDefaultDashboard = (data, setSelectedDashboard) => {
 export const DashboardReportTabView = ({ entityCode, TabSelector }) => {
   const [selectedYear, setSelectedYear] = useState(DEFAULT_DATA_YEAR);
   const [selectedDashboard, setSelectedDashboard] = useState(DEFAULT_DASHBOARD_GROUP);
+  const [tabBarHeight, setTabBarHeight] = useState(0);
+  const [isScrolledPastTop, setIsScrolledPastTop] = useState(false);
   const { data, isLoading, isError, error } = useDashboardData(entityCode);
 
+  const topRef = useRef();
+  const measureTabBarHeight = useCallback(tabBarNode => {
+    if (tabBarNode !== null) {
+      setTabBarHeight(tabBarNode.getBoundingClientRect().height);
+    }
+  }, []);
+  const stickyBarsHeight = useMemo(() => NAVBAR_HEIGHT_INT + tabBarHeight, [tabBarHeight]);
+
+  useEffect(() => {
+    const detectScrolledPastTop = () =>
+      setIsScrolledPastTop(topRef.current.getBoundingClientRect().top < stickyBarsHeight);
+
+    // detect once when the effect is run
+    detectScrolledPastTop();
+    // and again on scroll events
+    window.addEventListener('scroll', detectScrolledPastTop);
+
+    return () => window.removeEventListener('scroll', detectScrolledPastTop);
+  }, [stickyBarsHeight]);
+
+  const scrollToTop = useCallback(() => {
+    // if the top of the dashboards container is above the sticky dashboard header, scroll to the top
+    if (isScrolledPastTop) {
+      topRef.current.scrollIntoView();
+      window.scrollTo({ top: window.pageYOffset - stickyBarsHeight, behavior: 'smooth' });
+    }
+  }, [isScrolledPastTop, stickyBarsHeight]);
```

Comment:
A bunch of this stuff may need to be pulled into a more generic location when we have multiple dashboards

Reply:
Yeah that sounds good. It makes sense to keep the code in here at the moment since it's custom behaviour relating to this component.